README.md

```diff
@@ -1,87 +1,121 @@
+* `form_for` gives the option to dynamically change the `submit` button text (this comes in very handy when you're using a form partial and the `new` and `edit` pages will share the same form, but more on that in a later lesson)
 
-that magic line generates the following html for our edit form.
+A good rule of thumb for when to use one approach over the other is below:
 
-we never had to mention @post or post_title or @post.title. everything was automagically generated by calling a method, text_field, on the form_builder object.
+* Use `form_for` when your form is directly connected to a model, extending our example from the introduction, this would be our Hamster's profile edit form that connects to the profile database table
 
-lets break down the line that generates the text_field for the post titles.
+* User `form_tag` when you simply need an HTML form generated, examples of this would be: a search form field or a contact form
 
-form_builder.text_field is a method that form_builder responds too, we can even look it up in documentation.
 
-it's required argument is the attribute name of @post that we want to generate a text_field for, namely, the title attribute. That's all we have to say. The form builder knows what to do from there.
+## Implementation of `form_for`
 
-let's add a submit button to the form
+Let's take the `edit` form that utilized the `form_tag` that we built before for `posts` and refactor it to use `form_for`. As a refresher, here is the `form_tag` version:
 
-form_for(@post) do |form_builder| <%= form_builder.text_field :title %> <%= form_builder.submit %> end
+```ERB
+<% # app/views/posts/edit.html.erb %>
+<h3>Post Form</h3>
 
-refersh your browser. notice something amazing? we never specified the value or text of the submit button yet hte button knows we're updating a post? how? the magic form builder.
+<%= form_tag post_path(@post), method: "put" do %>
+  <label>Post title:</label><br>
+  <%= text_field_tag :title, @post.title %><br>
 
-lets go back to update and add a line to raise params.inspect because our form_builder has named our fields differently.
+  <label>Post Description</label><br>
+  <%= text_area_tag :description, @post.description %><br>
 
-we used to have post_title now we have post[title]. what could possibly be the meaning of this? this is just a string value for an html value, so nothing about 'post[title]' is ruby, but there is a magical rails convention here, what does post[title] look like in terms of ruby? what do you think the sturcture of params will be now?
+  <%= hidden_field_tag :authenticity_token, form_authenticity_token %>
+  
+  <%= submit_tag "Submit Post" %>
+<% end %>
+```
 
-lets submit the form and see
+Let's take this refactor one element at a time. Since we already have access to the `@post` instance variable we know that we can pass that to the `form_for` method, we also can remove the path argument and the `method` call, since `form_for` will automatically set these for us:
```

Comment:
Mention that since `@post` is an existing record, `form_for` knows that the method should be PUT